v0/destinations/bqstream/transform.js

```diff
@@ -42,4 +50,96 @@ const process = async event => {
   };
 };
 
-module.exports = { process };
+const batchEvents = eventsChunk => {
+  const batchedResponseList = [];
+
+  // arrayChunks = [[e1,e2, ..batchSize], [e1,e2, ..batchSize], ...]
+  const arrayChunks = _.chunk(eventsChunk, MAX_ROWS_PER_REQUEST);
+
+  // list of chunks [ [..], [..] ]
+  arrayChunks.forEach(chunk => {
+    const batchResponseList = [];
+    const metadata = [];
+
+    let batchEventResponse = defaultBatchRequestConfig();
+    const { message, destination } = chunk[0];
+
+    // Batch event into dest batch structure
+    chunk.forEach(ev => {
+      // Pixel code must be added above "batch": [..]
+      batchResponseList.push(ev.message.properties);
+      metadata.push(ev.metadata);
+    });
+
+    batchEventResponse.batchedRequest = {
+      datasetId: message.datasetId,
+      tableId: message.tableId,
+      projectId: message.projectId,
+      properties: batchResponseList
+    };
+
+    batchEventResponse = {
+      ...batchEventResponse,
+      metadata,
+      destination
+    };
+
+    batchedResponseList.push(
+      getSuccessRespEvents(
+        batchEventResponse.batchedRequest,
+        batchEventResponse.metadata,
+        batchEventResponse.destination,
+        true
+      )
+    );
+  });
+
+  return batchedResponseList;
+};
+
+const processRouterDest = async inputs => {
```

Comment:
There is already a utility for router transformation(without batching) in `v0/utils/index.js`. Can we see if we can make use of it ?

Reply:
updated